scipy/optimize/tests/test_differentiable_functions.py

```diff
@@ -326,6 +326,13 @@ def f(x):
         assert_equal(f2, 14.0)
         assert x is not sf.x
 
+    def test_non_scalar_func(self):
+        # Use of a non-scalar function should raise an error when using
+        # ScalarFunction
+        with assert_raises(ValueError):
```

Comment:
The newer `pytest` syntax is a bit nicer IMO:
```suggestion
        with pytest.raises(ValueError, match='must return a scalar):
```

Reply:
Ack, changed.